sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -336,6 +418,486 @@ private void CloseLink(ReceivingAmqpLink link)
+                return DisposeMessageRequestResponseAsync(
+                    lockTokens,
+                    timeout,
+                    DispositionStatus.Defered,
+                    _isSessionReceiver,
+                    SessionId,
+                    propertiesToModify);
+            }
+            return DisposeMessagesAsync(lockTokens, GetDeferOutcome(propertiesToModify), timeout);
+        }
+
+        /// <summary>
+        /// Abandons a <see cref="ServiceBusReceivedMessage"/>. This will make the message available again for processing.
+        /// </summary>
+        ///
+        /// <param name="message">The <see cref="ServiceBusReceivedMessage"/> to abandon.</param>
+        /// <param name="propertiesToModify">The properties of the message to modify while abandoning the message.</param>
+        /// <param name="cancellationToken"></param>
+        ///
+        /// <remarks>
+        /// Abandoning a message will increase the delivery count on the message.
+        /// This operation can only be performed on messages that were received by this receiver
+        /// when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
+        /// </remarks>
+        public override async Task AbandonAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object> propertiesToModify = null,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(ServiceBusReceiver));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            try
+            {
+                await _retryPolicy.RunOperation(
+                    async (timeout) => await AbandonInternalAsync(message, timeout, propertiesToModify).ConfigureAwait(false),
+                    EntityName,
+                    ConnectionScope,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                // MessagingEventSource.Log.MessageAbandonException(ClientId, exception);
+                throw exception;
+            }
+            finally
+            {
+                // diagnosticSource.DisposeStop(activity, lockToken, abandonTask?.Status);
+            }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            // MessagingEventSource.Log.MessageAbandonStop(ClientId);
+        }
+
+        /// <summary>
+        /// Abandons a <see cref="ServiceBusMessage"/> using a lock token. This will make the message available again for processing.
+        /// </summary>
+        ///
+        /// <param name="message">The lock token of the corresponding message to abandon.</param>
+        /// <param name="timeout"></param>
+        /// <param name="propertiesToModify">The properties of the message to modify while abandoning the message.</param>
+        internal virtual Task AbandonInternalAsync(
```

Comment:
nit: We are not appending `async` in other  internal methods(`CompleteInternal`, `DeferInternal`... etc), for the consistency can we update this to `AbandonInternal` or add `async` in other methods?

Reply:
Updated

Reply:
With the silly `Async` would be dropped altogether. Or the synchronous methods suffixed with `Sync` 😃